### PR TITLE
Shorten dshot pulse width

### PR DESCRIPTION
**Pull request by andyp1per**
It is possible to construct valid dshot and neopixel packets using fewer DMA beats. This allows more accurate frequency resolution because a lower frequency is required and the prescaler needed to divide up the system clock frequency can be higher. In testing this allowed ArduCopter to successfully drive BlueJay 0.14 ESCs from an omnibusf4pro-bdshot whereas with the current code the frequency difference is too large and the BlueJay firmware does not recognize the dshot signal.

This shortens the dshot pulse to 8 ticks with 3/8 for 0 and 6/8 for 1
This shortens the neopixel pulse to 11 ticks with 4/11 for 0 and 9/11 for 1

This appears to allow BlueJay ESCs to work with ArduPilot which are quite timing sensitive

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/RCOutput.cpp [hunk 1]
```diff
@@ -2232,7 +2239,7 @@ void RCOutput::_set_profiled_clock(pwm_group *grp, uint8_t idx, uint8_t led)
     const uint8_t bit_length = 25;
     const uint8_t stride = 4;
     uint32_t *buf = grp->dma_buffer + (led * bit_length + pad_start_bits) * stride + idx;
-    const uint32_t BIT_1 = 7 * grp->bit_width_mul;
+    const uint32_t BIT_1 = NEOP_BIT_0_TICKS * grp->bit_width_mul;
```
Comment (IamPete1) [inline libraries/AP_HAL_ChibiOS/RCOutput.cpp:2242]: Maybe a different define for ProfiLEDs.

Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/RCOutput.cpp:2242]: @IamPete1 sure! Are you able to point me at a spec where it specifies the pulse widths?

Comment (IamPete1) [reply libraries/AP_HAL_ChibiOS/RCOutput.cpp:2242]: No, its not actually a pulse width based protocol. Its is clock and data. It uses the different pulse widths to delay the clock rising edge relative to the data one. It should be quite robust to timing due to this. Just don't want someone to change the neopixel widths and accidentally break ProfiLEDs. In theory we should also be able to significantly increase the clock speed too, but it didn't seem worth pushing it too far.

Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/RCOutput.cpp]: Fixed

**Comment by andyp1per** [comment]
Flight tested on a 3" miniquad (Tekko32 ESCs). Arming tested on a 7" (T-Motor F55 Pro II). Neopixel tested on a 7". All worked fine.

**Review by tridge — APPROVED** [review]

**Comment by rmackay9** [comment]
@andyp1per, did you say that you wanted this included in 4.2 as well?

**Comment by andyp1per** [comment]
@rmackay9 yes - but only once its had a few more people flying it

**Pull request merged**